### PR TITLE
CPLAT-13196 Add Flag For Core Check

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Motivation
  
To prepare for the 2.9 corecheck enforcing that the repo isn't already transitioned, we needed to update the codemod to allow for a dry run and short circuit if no new boilerplate is found.

## Changes
  
- Add `LatestBoilerplateVisitor`, which looks for the new factory syntax
  - Add tests
  - Add `check-for-transitioning` flag
  - When the flag is set:
    -  Use visitor to check all dart files for new boilerplate
    - If there is new boilerplate, add the `--fail-on-changes` flag
    - Skip the pubspec update check
    - Run the codemod with the new flag
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
      - [ ] Steps from PR author: 
        - [ ] After pulling down and activating this codemod, verify it works
          > Use the command `pub global run over_react_codemod:dart2_9_upgrade --check-for-transitioning`
          - [ ] Run the codemode with the new flag on a project without the new updates. This should return normally with no changes.
          - [ ] Run the codemod without the new flag (to update the repo)
          - [ ] Run the codemod with the new flag again. This should log that it found new instances of the boilerplate, search for changes, and return normally
          - [ ] Manually remove a boilerplate change (so that it's the old boilerplate again)
          - [ ] Run the codemod with the new flag. This should end with a severe log indicating that the process failed
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by greglittlefield-wf — APPROVED** [review]

**Review by sydneyjodon-wk — COMMENTED** [review]
Just one comment. Working on QAing...

File: test/dart2_9_suggestors/latest_boilerplate_visitor_test.dart [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+            UiFactory<FooProps> Foo = random(castUiFactory(_$Foo)); // ignore: undefined_identifier
+            ''';
+
+            final boilerplateVisitor = LatestBoilerplateVisitor();
+            parseString(content: src).unit.accept(boilerplateVisitor);
+
+            expect(boilerplateVisitor.detectedLatestBoilerplate, true);
+          });
+
+          test('with multiple factories', () {
```
[the remark this answers is not in the document]
Comment (joebingham-wk) [reply test/dart2_9_suggestors/latest_boilerplate_visitor_test.dart:207]: Added in af4b8e5!

**Review by sydneyjodon-wk — APPROVED** [review]
+10 🎉  (pending CI passing)

> - [x] Run the codemod with the new flag on a project without the new updates. This should return normally with no changes.
> - [x] Run the codemod without the new flag (to update the repo)
> - [x] Run the codemod with the new flag again. This should log that it found new instances of the boilerplate, search for changes, and return normally
> - [x] Manually remove a boilerplate change (so that it's the old boilerplate again)
> - [x] Run the codemod with the new flag. This should end with a severe log indicating that the process failed

**Comment by joebingham-wk** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**